apiserver/src/main/java/org/dependencytrack/resources/v1/AdvisoriesResource.java

```diff
@@ -0,0 +1,380 @@
+            final List<VulnerableSoftware> vsList = BovModelConverter.extractVulnerableSoftware(bov);
+
+            LOGGER.debug("Synchronizing vulnerability " + vuln.getVulnId());
+            final Vulnerability persistentVuln = qm.synchronizeVulnerability(vuln, false);
+            qm.synchronizeVulnerableSoftware(persistentVuln, vsList, Vulnerability.Source.CSAF);
+
+            advisory.addVulnerability(persistentVuln);
+        }
+
+        // Sync the document into the database, replacing an older version with the same
+        // combination of tracking ID and publisher namespace if necessary
+        qm.synchronizeAdvisory(advisory);
+
+        return Response.ok("File uploaded successfully: " + fileName).build();
+    }
+
+    @DELETE
+    @Path("/{advisoryId}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Deletes an advisory", description = "<p>Requires permission <strong>VULNERABILITY_ANALYSIS_UPDATE</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "204", description = "Advisory removed successfully"),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "404", description = "The advisory could not be found")
+    })
+    @PermissionRequired(Permissions.Constants.VULNERABILITY_ANALYSIS_UPDATE)
+    public Response deleteAdvisory(@PathParam("advisoryId") String advisoryId) {
+        try (QueryManager qm = new QueryManager()) {
+            final Advisory entity = qm.getObjectById(Advisory.class, advisoryId);
+            if (entity != null) {
+                qm.delete(entity);
+                return Response.status(Response.Status.NO_CONTENT).build();
+            } else {
+                return Response.status(Response.Status.NOT_FOUND)
+                        .entity("The advisory could not be found.").build();
+            }
+        }
+    }
+
+    @POST
+    @Path("/seen/{advisoryId}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Marks an advisory as seen", description = "<p>Requires permission <strong>VULNERABILITY_ANALYSIS_UPDATE</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Advisory marked as seen successfully"),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "404", description = "The advisory could not be found")
+    })
+    @PermissionRequired(Permissions.Constants.VULNERABILITY_ANALYSIS_UPDATE)
+    public Response markAdvisoryAsSeen(@PathParam("advisoryId") String advisoryId) {
+        try (QueryManager qm = new QueryManager()) {
+            final Advisory entity = qm.getObjectById(Advisory.class, advisoryId);
+            if (entity != null) {
+                entity.setSeen(true);
+                qm.persist(entity);
```

Comment:
When multiple persistence operations are performed, please wrap them in a `qm.runInTransaction` or `qm.callInTransaction` block to ensure atomicity.